SQL/beestation_schema.sql

```diff
@@ -153,6 +153,16 @@ CREATE TABLE IF NOT EXISTS `SS13_characters` (
 ) COLLATE='utf8mb4_general_ci' ENGINE=InnoDB;
 
 
+-- Dumping structure for table ss13tgdb.SS13_characters_long
+DROP TABLE IF EXISTS `SS13_characters_long`;
+CREATE TABLE `SS13_characters_long` (
+	`ckey` VARCHAR(64) NOT NULL COLLATE 'utf8mb4_general_ci',
+	`slot` INT(11) UNSIGNED NOT NULL,
+	`preference_tag` VARCHAR(255) NOT NULL COLLATE 'utf8mb4_general_ci',
+	`preference_value` MEDIUMTEXT NULL COLLATE 'utf8mb4_general_ci',
+	PRIMARY KEY (`ckey`, `slot`) USING BTREE,
```

Comment:
> Shouldn't the preference tag also be in the primary key?

Depends on how the data is queried. The point of the key is to improve indexing speeds. If you're **always** running queries that are looking for an exact match of the three, then it should the primary key. If you're occasionally running those queries, then a secondary key would be a good addtion. If you're never running those queries, then it doesn't need to be a key at all.

Reply:
Mm, thought on this and Bacon's right. You'd need the tag to also be keyed here. Since you're doing a long table, there would be multiple entries on a single ckey/slot.